cli/command/environment_test.go

```diff
@@ -0,0 +1,419 @@
+package command
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/quintilesims/layer0/api/job"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+)
+
+func TestEnvironmentCreate_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	command := NewEnvironmentCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+		"OS missing":       NewContext(t, Args{"env_name"}, nil),
```

Comment:
Isn't OS an optional flag with `linux` as default? That's what is was in develop

Reply:
It is - but `EnvironmentCreateRequest` does validate that OS field is populated which is the reason that I've also added this as a test case.

Reply:
hmm that seems to be a mistake w/ the api implementation. should we update that?

Reply:
Do you mean not check the OS property in the EnvironmentCreateRequest.Validate()? I wouldn't say it is a mistake from the `EnvironmentCreateRequest` perspective. Seems reasonable to me that it can validate on things it deems as required.

Reply:
yeah - if it's supposed to be optional with a default, we should not error there on validate (would also need to check if it's set in `environment.Create`)